Aspire Budgeting/Facilities/StateManager.swift

```diff
@@ -2,44 +2,64 @@
 //  StateManager.swift
 //  Aspire Budgeting
 //
-//  Created by TeraMo Labs on 11/9/19.
-//  Copyright © 2019 TeraMo Labs. All rights reserved.
-//
 
 import Foundation
 import os.log
 
-final class StateManager: ObservableObject {
-  enum State: Equatable {
-    case loggedOut
-    case verifiedGoogleUser
-    case authenticatedLocally
-    case localAuthFailed
-    case needsLocalAuthentication
-    case hasDefaultSheet
-  }
+enum AppState: Equatable {
+  case loggedOut
+  case authenticatedExternally
+  case authenticatedLocally
+  case localAuthFailed
+  case needsLocalAuthentication
+  case hasDefaultFile
+}
+
+protocol AppStateManager {
+  var currentState: AppState { get }
+  var currentStatePublished: Published<AppState> { get }
+  var currentStatePublisher: Published<AppState>.Publisher { get }
+  
+  var isLocalAuthNeeded: Bool { get }
+  var isLoggedOut: Bool { get }
+  
+  func hasDefaultFile()
+}
 
-  @Published var currentStatePublisher: State = .loggedOut
-  private(set) var currentState: State = .loggedOut
+final class StateManager: AppStateManager {
+
+  @Published var currentState: AppState = .loggedOut
+  private(set) var __currentState: AppState = .loggedOut
+  var currentStatePublished: Published<AppState> { _currentState }
+  var currentStatePublisher: Published<AppState>.Publisher { $currentState }
+  
+  var isLocalAuthNeeded: Bool {
+    __currentState == .needsLocalAuthentication ||
+      __currentState == .localAuthFailed
+  }
+  
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/fd43c30d9ee7d075625997a607bdb1dd002cf1a8/Aspire Budgeting/Facilities/StateManager.swift#L40"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>StateManager.swift:40</code></td>
    </tr>
  </tbody>
</table>